mesonbuild/compilers/detect.py

```diff
@@ -10,6 +10,8 @@
 from ..envconfig import BinaryTable
 from .. import mlog
 
+from .mixins.clang import _LANG_MAP as clang_LANG_MAP
```

Comment:
[CI dislikes this](https://github.com/mesonbuild/meson/actions/runs/9636885448/job/26575361638?pr=13289#step:4:315), happy to do something else (or update the list instead).

Reply:
Just do the imports inside the functions using them.

This is just testing we don't start importing loads more modules for meson projects that don't even add any languages and thus don't run the compiler detection (only import detect.py itself).